functions/New-DbaDatabase.ps1

```diff
@@ -0,0 +1,469 @@
+Default is 10MB
+
+.PARAMETER PrimaryFileMaxSize
+The maximum permitted size in MB for the Primary File.  If this is less the primary file size for the Model database, then the Model size
+will be used instead.
+
+.PARAMETER Force
+The force parameter will ignore some errors in the parameters and assume defaults.
+
+.PARAMETER Silent 
+Use this switch to disable any kind of verbose messages
+
+.NOTES 
+Original Author: Matthew Darwin (@evoDBA, naturalselectiondba.wordpress.com)
+Tags: Database
+	
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.LINK
+https://dbatools.io/New-DbaAgentJob
+
+.EXAMPLE   
+New-DbaDatabase -SqlInstance sql1 -DatabaseName 'TestDatabase' -UserDataFileSize 128 -UserDataFileMaxSize 1024 -UserDataFileGrowth 128 `
+-LogSize 128 -LogGrowth 128
+Minimum required parameters; creates a database named TestDatabase on instance sql1 with a user filegroup with a single file of 128MB
+
+.EXAMPLE
+New-DbaDatabase -SqlInstance sql1, sql2, sql3 -DatabaseName 'MultiDatabaseTest' -UserDataFileSize 20 -UserDataFileGrowth 20 `
+-LogSize 20 -LogGrowth 20
+Creates a database named MultiDatabaseTest on instances sql1,sql2 and sql3
+
+.EXAMPLE
+New-DbaDatabase -SqlInstance sql1 -DatabaseName 'NonDefaultLocationTest' -NumberFilesInUserFilegroup 2 -$UseDefaultFileLocations $False `
+-NonDefaultFileLocation "C:\DBATools" -NonDefaultLogLocation "C:\DBATools"
+Creates a database named NonDefaultLogLocation in the C:\DBATools directory, with 2 files in the user filegroup.
+
+#>
+
+    [cmdletbinding(SupportsShouldProcess = $true, ConfirmImpact = "Low")]
+    Param 
+    ( 
+    # set variables for the database
+            [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		    [Alias("ServerInstance", "SqlServer")]
+		    [DbaInstanceParameter[]]$SqlInstance
+          , [PSCredential]$SqlCredential  
+          , [parameter(Mandatory = $true)]
+            [string]$DatabaseName                      
+          , [double]$NumberOfFilesInUserFilegroup = 1
+
+    # optional db variables
+          , [boolean]$UseDefaultFileLocations = $true
+          , [string]$NonDefaultFileLocation
+          , [string]$NonDefaultLogLocation
+          , [string]$Collation
+          , [string]$RecoveryModel
+          , [string]$DatabaseOwner = "sa"
+
+    #set the user data size, maxsize and growth
+          , [double]$UserDataFileSize 
+          , [double]$USerDataFileMaxSize
+          , [double]$UserDataFileGrowth #use 0 for no growth
+
+    #set the log size and growth
+          , [double]$LogSize
+          , [double]$LogGrowth
+
+    #set the primary file size in MB (will be converted to kb later)
+          , [double]$PrimaryFileSize = 10
+          , [double]$PrimaryFileGrowth = 10
+          , [double]$PrimaryFileMaxSize = 100
+          		
+    #Switches
+          , [switch]$Force
+		  , [switch]$Silent
+    )
+
```

Comment:
Would be cool to provide a properties parameter that accepts a hash table of the properties that can be set, if folks want to override model database settings.